lib/caching/CachedMccClient.ts

```diff
@@ -30,30 +30,31 @@ let defaultCachedMccClientOptions: CachedMccClientOptions = {
 //             eventually returned and then proceed with (i)
 
 @Managed()
-export class CachedMccClient<T, B> {
+export class CachedMccClient {
   client: ReadRpcInterface;
   chainType: ChainType;
 
-  transactionCache: Map<string, Promise<T>>;
+  transactionCache: Map<string, Promise<ITransaction>>;
   transactionCleanupQueue: Queue<string>;
 
-  blockCache: Map<string | number, Promise<B>>;
+  blockCache: Map<string | number, Promise<IBlock>>;
   blockCleanupQueue: Queue<string>;
 
   settings: CachedMccClientOptions;
 
+  // counters
   inProcessing = 0;
   inQueue = 0;
   reqsPs = 0;
   retriesPs = 0;
 
   cleanupCheckCounter = 0;
 
-  constructor(chainType: ChainType, options?: CachedMccClientOptions) {
+  constructor(chainType: ChainType, options?: CachedMccClientOptions, forcedClient?: ReadRpcInterface) {
```

Comment:
Why is `forcedClient` not part of CachedMccClientOptions like `inheritClient`?
It would be cleaner ;)